src/main/java/com/ibm/cloud/sdk/core/util/GsonSingleton.java

```diff
@@ -52,6 +52,7 @@ private static void registerTypeAdapters(GsonBuilder builder) {
     // Date serializer and deserializer
     builder.registerTypeAdapter(Date.class, new DateDeserializer());
     builder.registerTypeAdapter(Date.class, new DateSerializer());
+    builder.registerTypeAdapter(byte[].class, new ByteArrayTypeAdapter());
```

Comment:
I don't recall; is another type (e.g. `File`) being used for the alternative cases where `format: binary`?

If not then I don't think this type adapter should be registered "globally" because there may be some `byte[]` that need to be mapped with `type: string, format: binary` as opposed to `format: base64` (which was why I'd originally suggested annotating the fields). I can't find anything about what JSON string is actually expected when using `format: binary` other than "any sequence of octets", which I guess could just be the corresponding UTF-8 characters.

I agree that registering an adapter globally is a nicer solution, but we just need to take care that we don't break the other format by doing it.

Reply:
It looks to me like `type: string, format: byte`, which the openAPI generator calls `ByteArray`, is the only type mapped to `byte[]`, so it seems safe to me to register this type adapter globally.

Reply:
Correct, here is the complete type mapping for the Java generator:
```
        typeMapping.put("BigDecimal", "BigDecimal");
        typeMapping.put("ByteArray", "byte[]");
        typeMapping.put("file", "InputStream");
        typeMapping.put("binary", "InputStream");
        typeMapping.put("boolean", "Boolean");
        typeMapping.put("date", "Date");
        typeMapping.put("DateTime", "Date");
        typeMapping.put("number", "Double");
        typeMapping.put("float", "Float");
        typeMapping.put("double", "Double");
        typeMapping.put("long", "Long");
        typeMapping.put("integer", "Long"); // Use only 64-bit integers
        typeMapping.put("map", "Map");
        typeMapping.put("array", "List");
        typeMapping.put("UUID", "String");
        typeMapping.put("string", "String");
        typeMapping.put("object", "Object");
        typeMapping.put("any", "Object");
```
the Java "byte[]" type is only used for the openapi ByteArray type (which refers to a schema with "type string/format byte".